tachyon/zk/plonk/circuit/column_key.h

```diff
@@ -156,6 +157,38 @@ class ColumnKey : public ColumnKeyBase {
   }
   bool operator!=(const ColumnKey& other) const { return !operator==(other); }
 
+  // This ordering is consensus-critical! The layouters rely on deterministic
+  // column orderings.
+  bool operator<(const ColumnKey& other) const {
+    if (type_ == other.type_) {
+      if (type_ == ColumnType::kInstance || type_ == ColumnType::kFixed) {
+        return index_ < other.index_;
+      } else if (type_ == ColumnType::kAdvice) {
+        return phase_ < other.phase_;
+      }
+    } else {
+      // Across column types, sort Instance < Advice < Fixed.
+      if (type_ == ColumnType::kInstance) {
+        if (other.type_ == ColumnType::kAdvice ||
+            other.type_ == ColumnType::kFixed) {
+          return true;
+        }
+      } else if (type_ == ColumnType::kAdvice) {
+        if (other.type_ == ColumnType::kInstance) {
+          return false;
+        } else if (other.type_ == ColumnType::kFixed) {
+          return true;
+        }
+      } else if (type_ == ColumnType::kFixed) {
+        if (other.type_ == ColumnType::kInstance ||
+            other.type_ == ColumnType::kAdvice) {
+          return false;
+        }
+      }
```

Comment:
Please change the order of elements of `ColumnType` like below

```suggestion
  // THE ORDER OF ELEMENTS ARE IMPORTANT!! DO NOT CHANGE!
  enum class ColumnType {
    kAny,
    kInstance,
    kAdvice,
    kFixed,
  };
```
and change the lines as follows.

```suggestion
  return static_cast<int>(type_) < static_cast<int>(other.type_);
```